src/masternode-sync.cpp

```diff
@@ -311,7 +311,7 @@ bool CMasternodeSync::SyncWithNode(CNode* pnode, bool isRegTestNet)
         if (pnode->HasFulfilledRequest("getspork")) return true;
         pnode->FulfilledRequest("getspork");
 
-        pnode->PushMessage(NetMsgType::GETSPORKS); //get current network sporks
+        g_connman->PushMessageWithVersion(pnode, INIT_PROTO_VERSION, NetMsgType::GETSPORKS); //get current network sporks
```

Comment:
Same as above, this message is sent after the handshake, shouldn't be using INIT_PROTO_VERSION.

Reply:
Because `SyncWithNode` is called from a thread other than the messagehandler thread, there actually isn't any guarantee that these two message pushes are done after the handshake is complete. thats the reason i've used `INIT_PROTO_VERSION` here

Reply:
hmm a race, could be but then all of the messages over the masternode sync process are under the same race umbrella and not only the getsporks message.

Side from that, two other points:

1) Peers will not answer any message until the handshake is completed (check `ProcessMessage()`, if there is no version message then the node will increase the ban score - misbehaving peer - and not process the message, line 5438 of main.cpp.).
Peers shouldn't request data to an initialized peer.

2) `INIT_PROTO_VERSION` is not a wildcard version that the peer can send at any time, the remote side will not understand what is going on if it's send after the handshake (different versions).

What we can do to fix the possible race is skip the peers that haven't set the version yet on this thread and done (we will have peers at this point anyway, the tier two sync only starts when the chain is synced).

So, something like this can fix it:
```
g_connman->ForEachNodeContinueIf([sync, isRegTestNet](CNode* pnode){
        // skip nodes that haven't finished the handshake yet.
        if (pfrom->nVersion == 0) { return true; }
        return sync->SyncWithNode(pnode, isRegTestNet);
    });
```
(`nVersion` is atomic, so no problem on accessing here)